@here/harp-mapview/lib/Watermark.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as THREE from "three";
+
+/**
+ * Creates and handles rendering of watermark logo geometry for [[MapView]].
+ */
+export class Watermark {
+    private m_watermarkMesh: THREE.Mesh;
+    private readonly WATERMARK_RENDERORDER = 100000;
+    private readonly m_watermarkScene = new THREE.Scene();
+
+    /**
+     * Default constructor.
+     *
+     * @param textureUrl URL to watermark image.
+     * @param m_size Watermark size.
+     * @param m_margin Distance from canvas corner to watermark logo.
+     */
+    constructor(textureUrl: string, private m_size = 32, private m_margin = 4) {
+        const geometry = new THREE.PlaneGeometry(this.m_size, this.m_size);
+        const map = new THREE.TextureLoader().load(textureUrl);
```

Comment:
No handling of update after texture is actually loaded.
In case of remote images and static map this can cause that watermark will be never displayed.

OTOH, giving that watermark is kind-of business/legal measure it may be required that watermark is always displayed, so in this case we should wait with first render until watermark is really loaded.

Should it be even possible to  display map without watermark if it is configured ?